Different compiler needed on OpenBSD

On OpenBSD you need to install a newer compiler to get Nokogiri 1.6.8 to compile. This newer compiler is called `egcc` (for external I think). It would be great if Nokogiri's install process could detect it, so a normal `bundle install` would run without problems. Currently we need to manually connect to the server and run `CC=egcc bundle install`. Would you be willing to add such an exception/workaround? If so, what would be the best place to put this? I can then make a pull request for it.

Thanks and keep up the good work!

The problem is simple: default gcc in OpenBSD is too old to compile libxml2. It gives this error during install of nokogiri:

```
Extracting libxml2-2.9.4.tar.gz into tmp/x86_64-unknown-openbsd6.0/ports/libxml2/2.9.4... OK
Running 'configure' for libxml2 2.9.4... OK                                     
Running 'compile' for libxml2 2.9.4... ERROR, review '/home/frank/.gem/ruby/2.2/gems/nokogiri-1.6.8/ext/nokogiri/tmp/x86_64-unknown-openbsd6.0/ports/libxml2/2.9.4/compile.log' to see what happened. Last lines are:
========================================================================        
/usr/include/unistd.h:404: warning: redundant redeclaration of 'ftruncate'      
/usr/include/sys/types.h:217: warning: previous declaration of 'ftruncate' was here
/usr/include/unistd.h:429: warning: redundant redeclaration of 'truncate'       
/usr/include/sys/types.h:218: warning: previous declaration of 'truncate' was here
In file included from xmlIO.c:33:                                               
/usr/include/stdlib.h:184: warning: redundant redeclaration of 'mktemp'         
/usr/include/unistd.h:423: warning: previous declaration of 'mktemp' was here   
/usr/include/stdlib.h:215: warning: redundant redeclaration of 'mkstemp'        
/usr/include/unistd.h:422: warning: previous declaration of 'mkstemp' was here  
/usr/include/stdlib.h:255: warning: redundant redeclaration of 'mkdtemp'        
/usr/include/unistd.h:504: warning: previous declaration of 'mkdtemp' was here  
/usr/include/stdlib.h:295: warning: redundant redeclaration of 'mkstemps'       
/usr/include/unistd.h:505: warning: previous declaration of 'mkstemps' was here 
xmlIO.c: In function 'xmlXzfileClose':                                          
xmlIO.c:1450: error: 'LZMA_OK' undeclared (first use in this function)          
xmlIO.c:1450: error: (Each undeclared identifier is reported only once          
xmlIO.c:1450: error: for each function it appears in.)                          
*** Error 1 in . (Makefile:1248 'xmlIO.lo': @echo "  CC      " xmlIO.lo;/bin/sh ./libtool --silent --tag=CC    --mode=compile cc -DHAVE_CONF...)
*** Error 1 in . (Makefile:1465 'all-recursive')                                
*** Error 1 in /home/frank/.gem/ruby/2.2/gems/nokogiri-1.6.8/ext/nokogiri/tmp/x86_64-unknown-openbsd6.0/ports/libxml2/2.9.4/libxml2-2.9.4 (Makefile:859 'all')
========================================================================        
*** extconf.rb failed ***                        
```
with CC=egcc it works fine. I want to fix this in the gem by detecting OpenBSD and using the newer compiler.

--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -391,7 +391,7 @@ when arg_config('--clean')
   do_clean
 end
 
-if openbsd?
+if openbsd? && !using_system_libraries?
   ENV['CC'] ||= find_executable('egcc') or
     abort "Please install gcc 4.9+ from ports using `pkg_add -v gcc`"
 end

